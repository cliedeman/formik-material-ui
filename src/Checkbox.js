/* @flow */
import Checkbox from '@material-ui/core/Checkbox';

import createComponent from './createComponent';

export default createComponent(
  Checkbox,
  ({ field, form: { isSubmitting }, disabled = false, ...props }) => ({
    disabled: isSubmitting || disabled,
    ...props,
    ...field,
    // TODO: is this the correct way?
    value: field.value ? 'checked' : '',
  })
);
