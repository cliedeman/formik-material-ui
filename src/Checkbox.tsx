// @ts-ignore
import * as React from 'react';
import MuiCheckbox, {
  CheckboxProps as MuiCheckboxProps,
} from '@material-ui/core/Checkbox';
import { FieldProps } from 'formik';
import { Omit } from './types';

export interface CheckboxProps extends FieldProps, Omit<MuiCheckboxProps, 'form'> {}

const Checkbox: React.ComponentType<CheckboxProps> = ({
  field,
  form: { isSubmitting },
  disabled = false,
  ...props
}) => (
  <MuiCheckbox
    disabled={isSubmitting || disabled}
    {...props}
    {...field}
    // TODO: is this the correct way?
    value={field.value ? 'checked' : ''}
  />
);

Checkbox.displayName = 'FormikMaterialUICheckbox';

export default Checkbox;
