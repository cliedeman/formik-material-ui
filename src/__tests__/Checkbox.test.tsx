/* @flow */
import React from 'react';
import { Formik, Field, Form } from 'formik';
import renderer from 'react-test-renderer';

import Checkbox from '../Checkbox';

test('Checkbox Renders Correctly', () => {
  const component = renderer.create(
    <Formik initialValues={{ test: true }} onSubmit={() => null}>
      <Form>
        <Field name="test" label="Checkbox Label" component={Checkbox} />
      </Form>
    </Formik>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
