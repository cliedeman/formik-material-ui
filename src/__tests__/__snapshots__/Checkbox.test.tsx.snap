// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Renders Correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <span
    className="MuiButtonBase-root-17 MuiIconButton-root-11 MuiSwitchBase-root-7 MuiCheckbox-root-1 MuiCheckbox-colorSecondary-6"
    label="Checkbox"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-16"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-20"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <input
        checked=""
        className="MuiSwitchBase-input-10"
        data-indeterminate={false}
        disabled={false}
        name="test"
        onChange={[Function]}
        type="checkbox"
        value=""
      />
    </span>
    <span
      className="MuiTouchRipple-root-29"
    />
  </span>
</form>
`;
